atspi-common/src/events/mod.rs

```diff
@@ -170,12 +152,41 @@ impl Default for EventBodyOwned {
 			kind: String::new(),
 			detail1: 0,
 			detail2: 0,
-			any_data: Value::U8(0u8).into(),
+			any_data: 0u8.into(),
 			properties: HashMap::new(),
 		}
 	}
 }
 
+// Manual implementation to account for the remote chance that either `any_data` or `properties`
+// fail to clone because the clone would exceed the open file limit.
+impl Clone for EventBodyOwned {
+	fn clone(&self) -> Self {
+		let cloned_any_data = self.any_data.try_clone().unwrap_or_else(|err| {
+			panic!("Failure cloning 'any_data' field: {err:?}");
+		});
+
+		let cloned_properties = {
+			let mut map = HashMap::new();
+			for (key, value) in &self.properties {
+				let cloned_value = value.try_clone().unwrap_or_else(|err| {
+					panic!("Failure cloning 'props' field: {err:?}");
+				});
+				map.insert(key.clone(), cloned_value);
+			}
+			map
+		};
+
+		Self {
+			kind: self.kind.clone(),
+			detail1: self.detail1,
+			detail2: self.detail2,
+			any_data: cloned_any_data,
+			properties: cloned_properties,
+		}
+	}
+}
+
```

Comment:
Should we just let the users of atspi call `.try_clone()` instead? And let them decide whether or not to fail?

Reply:
I decided to be 'lust practical' considering the context:

Did or will `EventBodyOwned.any_data` ever contain a file?
Did or will `EventBodyOwned.properties` ever contain a file?

I don't think so and I think the chances are slim they ever will and should they ever, the chances of triggering the failure path is slim.

It expected that a type has `Clone`.